### PR TITLE
INTMDB-247: Fixes a bug where it's taking 3 minutes to read a cluster

### PR DESCRIPTION
**Pull request by coderGo93**
## Description

- Deleted the delay of 3 minutes which is the reason why it "hangs" in order to read a cluster with datasource
- Tested manually but modified the terratest to verify it works
- In tests, if you create with private link it should take to read from 1:30min to 3min at most, if you use directly datasource to get other values, it should be immediately to get a result
Results:

Original (with 3 minutes delay)

```
mongodbatlas_privatelink_endpoint_service.atlaseplink: Creation complete after 2m13s [id=ZW5kcG9pbnRfc2VydmljZV9pZA==:dnBjZS0wOWNiNDMxNWNlNTQ1NDRiMQ==-cHJpdmF0ZV9saW5rX2lk:NjEyNTliM2UxZDZkMjc1OTg2ZDc3ZGE4-cHJvamVjdF9pZA==:NWNmNWE0NWE5Y2NmNjQwMGU2MDk4MWI2-cHJvdmlkZXJfbmFtZQ==:QVdT]
data.mongodbatlas_cluster.cluster-atlas: Reading...
data.mongodbatlas_cluster.cluster-atlas: Still reading... [10s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [20s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [30s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [40s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [50s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m0s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m10s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m20s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m30s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m40s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m50s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [2m0s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [2m10s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [2m20s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [2m30s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [2m40s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [2m50s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [3m0s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Read complete after 3m1s [id=61256b8ce70fbb7cf8ad6eee]

```

Actual (depending, it can take from 1:30 min to 3min to read a cluster after private link)
```
aws_vpc_endpoint.ptfe_service: Creation complete after 8s [id=vpce-0e3cb9799ae7f2238]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Creating...
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [10s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [20s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [30s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [40s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [50s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [1m0s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [1m10s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [1m20s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [1m30s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [1m40s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [1m50s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [2m0s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [2m10s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Still creating... [2m20s elapsed]
mongodbatlas_privatelink_endpoint_service.atlaseplink: Creation complete after 2m24s [id=ZW5kcG9pbnRfc2VydmljZV9pZA==:dnBjZS0wZTNjYjk3OTlhZTdmMjIzOA==-cHJpdmF0ZV9saW5rX2lk:NjEyNjY4MGE5YWMxMmUwM2I2NzgyMzFm-cHJvamVjdF9pZA==:NWNmNWE0NWE5Y2NmNjQwMGU2MDk4MWI2-cHJvdmlkZXJfbmFtZQ==:QVdT]
data.mongodbatlas_cluster.cluster-atlas: Reading...
data.mongodbatlas_cluster.cluster-atlas: Still reading... [10s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [20s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [30s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [40s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [50s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m0s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m10s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Still reading... [1m20s elapsed]
data.mongodbatlas_cluster.cluster-atlas: Read complete after 1m22s [id=61256b8ce70fbb7cf8ad6eee]

```

If you use directly for other purposes, the result should be immediately

Link to any related issue(s):
#521 
## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/data_source_mongodbatlas_cluster.go
Comment (themantissa) [inline mongodbatlas/data_source_mongodbatlas_cluster.go]: Is there a way to skip they delay unless absolutely needed like in the case of privatelink?  I mean if it would return in a second it doesn't seem to make sense to delay it arbitrarily.

Comment (coderGo93) [reply mongodbatlas/data_source_mongodbatlas_cluster.go]: Yes, I think it's possible to skip the delay, let me try and push the changes if it works

File: examples/aws-atlas-privatelink/atlas-cluster.tf
Comment (thetonymaster) [inline examples/aws-atlas-privatelink/atlas-cluster.tf]: Add a NL

Comment (coderGo93) [reply examples/aws-atlas-privatelink/atlas-cluster.tf]: done, thank you

File: mongodbatlas/data_source_mongodbatlas_cluster.go
Comment (coderGo93) [reply mongodbatlas/data_source_mongodbatlas_cluster.go]: Deleted the unnecessary delay

**Review by themantissa — APPROVED** [review]
LGTM

**Comment by gssbzn** [comment]
Tests are failing here

**Review by gssbzn — APPROVED** [review]
Not familiar enough (yet) with the tests so let's make sure there's not a regression from this change

**Review by thetonymaster — APPROVED** [review]

**Comment by coderGo93** [comment]
Tested manually and it worked.
User cases:
- If you create a cluster then datasource of cluster(without private endpoints), it work immediately
- If you create a cluster then datasource of cluster with private endpoints, it should get the result immediately

Has the approval, you can merge now @themantissa .

FYI: For some reason the normal test fails because it doesn't get the env var(secrets) even I set in the fork repo

**Comment by themantissa** [comment]
@coderGo93 deleted my comment as you posted while I was writing mine.   If the tests are failing because of env var we should figure that out before merging.   Please figure it out and once it passes I'll merge.

**Pull request merged**
